src/frontend/src/app/project/project.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, OnInit } from '@angular/core';
+import {ProjectDto} from "../model/projectDto";
+import {FormBuilder, FormGroup, Validators} from "@angular/forms";
+import {Router} from "@angular/router";
+import { ProjectService } from "./project.service";
+import { HttpClient } from "../shared/http.client.service";
+
+@Component({
+  selector: 'app-project',
+  templateUrl: './project.component.html',
+  styleUrls: ['./project.component.css']
+})
+export class ProjectComponent implements OnInit {
+
+  projectDto: ProjectDto = new ProjectDto();
+  projectForm: FormGroup;
+  error = '';
+
+  constructor(fb: FormBuilder, private router: Router, private projectService: ProjectService, httpClient: HttpClient) {
+
+    this.projectForm = fb.group({
+      name: [null, [Validators.required, Validators.minLength(5), Validators.maxLength(30)]],
+      description: [null, [Validators.required, Validators.minLength(5), Validators.maxLength(300)]],
+      projectKey: [null, [Validators.required, Validators.minLength(3), Validators.maxLength(8)]]
+    });
+
+  }
+
+  ngOnInit() {
+  }
+
+  createproject(){
+    this.projectService.createProject(this.projectDto)
+      .subscribe(
+        success => {
+          this.router.navigate(['/home']);
+        },
+        error => {
+          if(error.status === 409){
+            this.error = 'Project with key ' + this.projectDto.projectKey + ' already exists!';
```

Comment:
na back endzie masz zdefiniowany exception z takim samym komunikatem. żeby pobrać komunikat z responsa użyj tutaj error._body

Reply:
this.error = error._body